src/components/Sliding/index.spec.js

```diff
@@ -0,0 +1,114 @@
+import { describe, expect, it } from 'vitest';
+import { render, screen } from '@testing-library/vanilla';
+import Slinding from '.';
+
+const $slide1 = document.createElement('div');
+$slide1.style.height = '200px';
+$slide1.style.backgroundColor = 'red';
+$slide1.textContent = 'slide 1';
+
+const $slide2 = document.createElement('div');
+$slide2.style.height = '200px';
+$slide2.style.backgroundColor = 'pink';
+$slide2.textContent = 'slide 2';
+
+const $slide3 = document.createElement('div');
+$slide3.style.height = '200px';
+$slide3.style.backgroundColor = 'green';
+$slide3.textContent = 'slide 3';
+
+const $slide4 = document.createElement('div');
+$slide4.style.height = '200px';
+$slide4.style.backgroundColor = 'black';
+$slide4.textContent = 'slide 4';
+
+const propsMock = {
+  slides: [$slide1, $slide2, $slide3],
+};
+
+const makeSut = (parameters) => render(new Slinding(parameters));
+
+describe('Sliding', () => {
+  describe('when mount', () => {
+    it('render with 3 itens', async () => {
+      makeSut(propsMock);
+      const slide1 = await screen.findByText('slide 1');
+      const slide2 = await screen.findByText('slide 2');
+      const slide3 = await screen.findByText('slide 3');
+
+      expect(slide1).toBeInTheDocument();
+      expect(slide2).toBeInTheDocument();
+      expect(slide3).toBeInTheDocument();
+    });
+  });
+
+  describe('when call functions', () => {

```

Comment:
poderia ser `method`, mas eu acho que não gosto desse `describe` aqui nao. Vc precisa ser capaz de ler os describes até o it de uma forma linear, por exemplo
`Slide method add item programmatically`, no caso eu não sei bem oq adicionar nesse describe aqui em cima, mas é valido ver oq pode adicionar.